source/common/common/logger.h

```diff
@@ -315,7 +315,7 @@ class Context {
   const std::string log_format_;
   Thread::BasicLockable& lock_;
   bool should_escape_;
-  bool enable_fine_grain_logging_;
+  std::atomic<bool> enable_fine_grain_logging_;
```

Comment:
Is there any performance penalty for std::atomic? since this will be read for every log
https://github.com/envoyproxy/envoy/blob/4d57ba2805458b9141e01a7e293857ed2fc7fd5a/source/common/common/logger.h#L474

Reply:
+1 to benchmark any such change